### PR TITLE
vim-patch:8.1.{140,1269},8.2.0161

### PR DESCRIPTION
**Pull request by janlazo**

### CODE REVIEW

**Comment by justinmk** [comment]
One test failure, but maybe the new behavior is correct?

```
[  FAILED  ] test/functional/terminal/mouse_spec.lua @ 186: :terminal mouse when the terminal has focus with a split window and other buffer handles terminal size when switching buffers
test/functional/terminal/mouse_spec.lua:199: Row 1 did not match.
Expected:
  |*rows: 5, cols: 25        │rows: 5, cols: 25       |
  |*rows: 5, cols: 24        │rows: 5, cols: 24       |
  |*mouse enabled            │mouse enabled           |
  |*rows: 5, cols: 25        │rows: 5, cols: 25       |
  |*{2:^ }                        │{2: }                       |
  |==========                ==========              |
  |:bn                                               |
Actual:
  |*^line21                   │rows: 5, cols: 25       |
  |*line22                   │rows: 5, cols: 24       |
  |*line23                   │mouse enabled           |
  |*line24                   │rows: 5, cols: 25       |
  |*line25                   │{2: }                       |
  |==========                ==========              |
  |:bn                                               |
```

**Comment by janlazo** [comment]
8.1.0425 was suppose to fix 8.1.0342 which fixes 8.0.1732. I expected 0 test failures 

I'll leave 8.1.0425 WIP until the other (terminal) patches are merged. Need to resolve https://github.com/neovim/neovim/pull/10396 before continuing on the other terminal patches.

8.2.0161 is safe to cherry-pick.

File: src/nvim/keymap.h [hunk 1]
```diff
@@ -240,8 +240,8 @@ enum key_extra {
   , KE_DROP = 95              // DnD data is available
   // , KE_CURSORHOLD = 96     // CursorHold event
   , KE_NOP = 97               // no-op: does nothing
-  , KE_FOCUSGAINED = 98       // focus gained
-  , KE_FOCUSLOST = 99         // focus lost
+  // , KE_FOCUSGAINED = 98    // focus gained
+  // , KE_FOCUSLOST = 99      // focus lost
```
Comment (janlazo) [inline src/nvim/keymap.h]: These are used in the original vim patch for `gotchars()` but Neovim removed `K_FOCUS` pseudo-keys.

File: src/nvim/getchar.c [hunk 2]
```diff
@@ -1096,26 +1096,40 @@ void del_typebuf(int len, int offset)
...
+    buf[buflen++] = *s++;
+
+    // When receiving a special key sequence, store it until we have all
+    // the bytes and we can decide what to do with it.
+    if (buflen == 1 && buf[0] == K_SPECIAL) {
+      continue;
+    }
+    if (buflen == 2) {
+      continue;
+    }
```
Comment (janlazo) [inline src/nvim/getchar.c]: Omitted the `buflen == 3` condition because Neovim doesn't support `K_FOCUS` pseudo keys.

**Review by justinmk — APPROVED** [review]
waiting on CI

**Pull request merged**